### ISSUE TITLE
Optimise v_ramp by converting it into a functional component

### ISSUE THREAD

**Issue by gerhean** [reposense/RepoSense#1391]
**What feature(s) would you like to see in RepoSense?**

Over a thousand v_ramp components can be rendered on a larger page. Let's optimize it by converting it into a functional component.

**Is the feature request related to a problem?**

Need more testing but since so many v_ramp is rendered it might be one of the reasons why the site loads slowly.

**If possible, describe the solution**

According to https://medium.com/js-dojo/vue-js-functional-components-what-why-and-when-439cfaa08713, functional components can increase rendering performance by 70%.

Looking at `v_ramp.js`, we realize that it does not need to be stateful.

**If applicable, describe alternatives you've considered**

Upgrading to Vue3 is said to massively improve performance of stateless components without any need to implement functional components

**Additional context**

A prototype implementation shows some performance gain:

![image](https://user-images.githubusercontent.com/16630400/103408837-3110cd00-4b9f-11eb-84ed-6119144e2e49.png)

vs the current implementation:

![image](https://user-images.githubusercontent.com/16630400/103408867-42f27000-4b9f-11eb-87b7-4767d45fd63c.png)

Done by recording performance when changing the group by dropdown. Not tested super rigorously though so might be a fluke.

**Comment by fzdy1914** [comment]
Can have a try, seems promising.

**Comment by Zhou-Jiahao-1998** [comment]
@gerhean 
Quoting from [Vue document](https://v3.vuejs.org/guide/migration/functional-components.html#_2-x-syntax): 
```
Performance gains from 2.x for functional components are now negligible in 3.x,
so we recommend just using stateful components
```

Furthermore, the Vue 2.x syntax will break in 3.x. Therefore, I think this might not
worth the time and effort. This issue will be solved if we eventually upgrade to Vue 3.

**Comment by gerhean** [comment]
We can't say for sure until we upgraded to Vue3 and tested this. However I will say that the code for functional components is somewhat hard to read and inconsistent with the rest of the code base, which was the reason I had abandoned this issue despite the increase in performance. Vue3 might have something to declare a component as static without making it functional, I'm not sure.

I think this issue can be closed, but the long render time of the ramps is still a very big concern and needs to be optimised.
